x/cdp/keeper/querier_test.go

```diff
@@ -303,6 +303,17 @@ func (suite *QuerierTestSuite) TestQueryAccounts() {
 	suite.Require().True(findByName("savings"))
 }
 
+func (suite *QuerierTestSuite) TestQuerySavingsRateDistributed() {
+	ctx := suite.ctx.WithIsCheckTx(false)
+	bz, err := suite.querier(ctx, []string{types.QueryGetSavingsRateDistributed}, abci.RequestQuery{})
+	suite.Nil(err)
+	suite.NotNil(bz)
+
+	var distAmount sdk.Int
+	suite.Nil(types.ModuleCdc.UnmarshalJSON(bz, &distAmount))
+	suite.True(sdk.ZeroInt().Equal(distAmount))
+}
+
 func TestQuerierTestSuite(t *testing.T) {
```

Comment:
Given complexity of the matching/filtering cdp function, we should have a test for `QueryGetCdps`. Either a new test in the querier suite or even a unit test of the `filterCDPs` and `findItersection` functions

Reply:
Added tests for `FindIntersection`, `FilterCDPs`, and `QueryGetCDPs` in https://github.com/Kava-Labs/kava/pull/644/commits/2f1cde9217553fff9a071300d1b0092291a36024.

Had to capitalize some functions so I could import them into the test suite. Wonder what the 'proper' approach to module/module test organization would be here, exposing internal functions just for the test suite feels off.

Reply:
Yeah I've run into this as well. There didn't seem to be a consensus best answer, so I've generally resorted to making functions public for purposes of testing, which feels bad.

Reply:
Anything that requires a keeper/app to be setup needs to be exported. If I remember correctly the test files need be in a `*_test` package to avoid import cycles.
But anything that doesn't require a keeper/app can live in a normal test file. Tests for `findIntersection` and `filterCDP`could live in a `filter_test.go` or something with `package keeper`. Though not important